test/RandomNumberConsumer_test.js

```diff
@@ -1,29 +1,55 @@
-const { expect } = require('chai')
+const chai = require('chai')
+const {expect} = require('chai')
+const BN = require('bn.js');
+chai.use(require('chai-bn')(BN));
 
 describe('RandomNumberConsumer', async function () {
+  
   let randomNumberConsumer, vrfCoordinatorMock, seed
-
+  
   beforeEach(async () => {
-    await deployments.fixture(['mocks', 'vrf'])
+     //if local chain, then we use mocks
+     if (await getChainId() == 31337) {
+      await deployments.fixture(['mocks', 'vrf'])
+      const LinkToken = await deployments.get('LinkToken')
+      linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
+      const VRFCoordinatorMock = await deployments.get('VRFCoordinatorMock')
+      vrfCoordinatorMock = await ethers.getContractAt('VRFCoordinatorMock', VRFCoordinatorMock.address)
+    }
 
+    // We always need to set a seed and set the RandomNumberConsumer contract
     seed = 123
-    const LinkToken = await deployments.get('LinkToken')
-    linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
     const RandomNumberConsumer = await deployments.get('RandomNumberConsumer')
     randomNumberConsumer = await ethers.getContractAt('RandomNumberConsumer', RandomNumberConsumer.address)
-    const VRFCoordinatorMock = await deployments.get('VRFCoordinatorMock')
-    vrfCoordinatorMock = await ethers.getContractAt('VRFCoordinatorMock', VRFCoordinatorMock.address)
+    
   })
 
   it('Should successfully make an external random number request', async () => {
-    const expected = '777'
-    await linkToken.transfer(randomNumberConsumer.address, '2000000000000000000')
-    const transaction = await randomNumberConsumer.getRandomNumber(seed)
-    const tx_receipt = await transaction.wait()
-    const requestId = tx_receipt.events[2].topics[0]
+    // If local chain, do tests using mocks
+    if (await getChainId() == 31337) {
```

Comment:
same comment here about having two massive code blocks that are really only have one or two lines different.